sweetest/src/main/java/com/mysugr/sweetest/framework/context/DependenciesTestContext.kt

```diff
@@ -112,33 +121,34 @@ class DependenciesTestContext {
      */
     private fun addDependency(clazz: KClass<*>, block: (DependencyState<out Any>, DependencyMode?) -> Unit) {
         with(TestEnvironment.dependencies) {
-            assertDependency(clazz)
             val dependencyState = getDependencyConfiguration(clazz)?.let { configuration ->
-                states[configuration]
+                return@let states.getByConfiguration(configuration)
             } ?: run {
-                states[DependencyConfiguration(clazz, null, null, null)]
+                val configuration = DependencyConfiguration(clazz, null, null, null)
+                return@run states.getByConfiguration(configuration)
             }
-
             block(dependencyState, dependencyState.configuration.defaultDependencyMode)
         }
     }
 
     /**
-     * Asserts if the type [clazz] is already added.
+     * Asserts if the configured mode matches the requested one.
      */
-    private fun assertDependency(clazz: KClass<*>) {
-        check(TestEnvironment.dependencies.states[clazz] == null) {
-            "Can't change dependency mode or ${clazz.simpleName}, it " +
-                "has already been set before and can't be changed afterwards"
+    private fun assertConfiguredMode(clazz: KClass<*>, configurationMode: DependencyMode?, mode: DependencyMode) {
+        check(configurationMode == mode || configurationMode == null) {
+            "Dependency $clazz has been configured as ${configurationMode?.name} but is requested as ${mode.name}! "
```

Comment:
i think this message can still be a bit more conclusive. what about:

```suggestion
            "Dependency \"$clazz.simpleName\" has been forced to be ${configurationMode?.name} in the module testing configuration, but you requested it to be ${mode.name} instead. Please loosen up the constraint by using \"any\" instead of \"requireMock\" or \"requireReal\" or remove the module testing configuration entirely as it's deprecated anyway."
```